### DISCUSSION TITLE
De-selecting custom radio button?

### DISCUSSION THREAD

**Discussion by karrui** [chakra-ui/chakra-ui#4252] [Q&A] [answered]
I'm creating a YesNo component that is basically a radio group with two custom radios ("yes" option, "no" option). However, I would also like to allow users to deselect the Yes/No options (basically resetting the radio group to empty values). 

I have managed to make it work with an onClick handler that checks if the current `prop.isChecked` is true, and then running `inputProp`'s onChange handler with an empty value if it is. This has the effect of selecting nothing when the same radio option is clicked again:

```ts
const handleSelect = useCallback(() => {
  if (props.isChecked && input.onChange) {
    // eslint-disable-next-line @typescript-eslint/no-extra-semi
    ;(input.onChange as UseRadioGroupReturn['onChange'])('')
  }
}, [input.onChange, props.isChecked])

...

<input
  {...input}
  onClick={handleSelect}
/>
```

However, I cannot get it to work with Spacebar toggle (for a11y). 

I have tried running the handler on `onKeyUp` events like so:

```ts
<input
  {...input}
  onClick={handleSelect}
  onKeyUp={(e) => {
    if (e.key !== ' ') return
    if (props.isChecked) {
      e.preventDefault()
      handleSelect()
    }
  }}
/>
```

`e.preventDefault` must be called, or the default spacebar behaviour is to re-select the radio button right after it's toggled off and thus nothing actually happens (`handleSelect` toggles off, only to have the default behaviour toggle the radio selection again).

However, because `e.preventDefault()` is called, the `data-active` attribute is not removed and remains on the "toggled off" radio, resulting in poor a11y and styling (the `_active` styles continue to be applied)

**So the main question is**, 
1. is there a better way of doing this, and if not,
2. is there a way to remove data-active programatically?
3. is there a way to allow for all the spacebar default behaviour **EXCEPT** reselecting the radio?

**Comment by karrui** [comment, marked as answer]
Seconds after I've posted this.... I tried the `onKeyDown` event instead and everything works perfectly:

```ts
<input
          {...input}
          onClick={handleSelect}
          onKeyDown={(e) => {
            if (e.key !== ' ') return
            if (props.isChecked) {
              e.preventDefault()
              handleSelect()
            }
          }}
        />
```

**Reply by dkt0591** [reply]
Sir, you saved my life!!!! im searching whole internet for thisssssss

**Reply by amariani** [reply]
Worked perfectly! Thanks @karrui
